python/lsst/sims/catalogs/generation/db/ObservationMetaData.py

```diff
@@ -60,70 +60,94 @@ class ObservationMetaData(object):
 
     """
     def __init__(self, boundType=None, boundLength=None,
-                 mjd=None, unrefractedRA=None, unrefractedDec=None, rotSkyPos=0.0,
-                 bandpassName='r', phoSimMetadata=None, site=None, m5=None, skyBrightness=None):
+                 mjd=None, unrefractedRA=None, unrefractedDec=None, rotSkyPos=None,
+                 bandpassName=None, phoSimMetaData=None, site=Site(), m5=None, skyBrightness=None):
 
-        self.bounds = None
-        self.boundType = boundType
-        self.mjd = mjd
-        self.bandpass = bandpassName
-        self.skyBrightness = skyBrightness
+        self._bounds = None
+        self._boundType = boundType
+        self._mjd = mjd
+        self._bandpass = bandpassName
+        self._skyBrightness = skyBrightness
+        self._site = site
 
         if rotSkyPos is not None:
-            self.rotSkyPos = numpy.radians(rotSkyPos)
+            self._rotSkyPos = numpy.radians(rotSkyPos)
         else:
-            self.rotSkyPos = None
+            self._rotSkyPos = None
 
         if unrefractedRA is not None:
-            self.unrefractedRA = numpy.radians(unrefractedRA)
+            self._unrefractedRA = numpy.radians(unrefractedRA)
         else:
-            self.unrefractedRA = None
+            self._unrefractedRA = None
 
         if unrefractedDec is not None:
-            self.unrefractedDec = numpy.radians(unrefractedDec)
+            self._unrefractedDec = numpy.radians(unrefractedDec)
         else:
-            self.unrefractedDec = None
+            self._unrefractedDec = None
 
         if boundLength is not None:
-            if isinstance(boundLength, float):
-                self.boundLength = numpy.radians(boundLength)
-            else:
-                try:
-                    self.boundLength = []
-                    for ll in boundLength:
-                        self.boundLength.append(numpy.radians(ll))
-                except:
-                    raise RuntimeError("You seem to have passed something that is neither a float nor " +
-                                       "list-like as boundLength to ObservationMetaData")
+            self._boundLength = numpy.radians(boundLength)
```

Comment:
If we want to have exceptions to prevent inputs of incorrect types from users, we can be more specific: it can be one of two things: 
a) a float
b) collections.Iterables instance with length of 2 containing floats. Also this only makes sense with boundType='box',

Maybe this is too much here